### PR TITLE
sanitize strings with shell control characters

### PR DESCRIPTION
**Pull request by Kriechi**
@mhils: anything else we should sanitize?

### CODE REVIEW

File: mitmproxy/dump.py [hunk 1]
```diff
@@ -295,7 +296,7 @@ def _echo_response_line(self, flow):
             replay=replay,
             arrows=arrows,
             code=code,
-            reason=reason,
+            reason=shlex_quote(reason),
```
Comment (mhils) [inline mitmproxy/dump.py]: we need to apply this before `click.style`

Comment (Kriechi) [inline mitmproxy/dump.py]: I changed it - please take another look.

File: netlib/utils.py [hunk 2]
```diff
@@ -431,3 +430,16 @@ def safe_subn(pattern, repl, target, *args, **kwargs):
         need a better solution that is aware of the actual content ecoding.
     """
     return re.subn(str(pattern), str(repl), target, *args, **kwargs)
+
+def dequote(s):
```
Comment (mhils) [inline netlib/utils.py]: This is very similar to pathod's `inner_repr`. Can we combine these?
(cc @dufferzafar)

Also, tests please.

File: netlib/utils.py [hunk 3]
```diff
@@ -431,3 +431,6 @@ def safe_subn(pattern, repl, target, *args, **kwargs):
         need a better solution that is aware of the actual content ecoding.
     """
     return re.subn(str(pattern), str(repl), target, *args, **kwargs)
+
+def bin_safe(s):
+    return ''.join(["\\x{:02x}".format(ord(i)) if ord(i) < 32 else i for i in s])
```
Comment (mhils) [inline netlib/utils.py]: This looks very much like `repr` now (we miss 127 though) - maybe just do `repr(x)` + strip outer parantheses?

Comment (Kriechi) [inline netlib/utils.py]: repr also might add `u` and quotes and stuff. I want keep the string as much as possible to the original version. Not sure how this all changes with bytes vs. str vs. unicode...

Comment (mhils) [inline netlib/utils.py]: We need something similar for pathod as well. I'll work on it.

**Pull request closed without merging**